src/Libraries/Microsoft.Extensions.Telemetry/Logging/ExtendedLoggerFactory.cs

```diff
@@ -43,6 +46,7 @@ public ExtendedLoggerFactory(
 #pragma warning restore S107 // Methods should not have too many parameters
     {
         _scopeProvider = scopeProvider;
+        _sampler = sampler ?? new AlwaysOnSampler();
```

Comment:
Performance-wise its probably a little bit faster to execute `_sampler == null ? true : _sampler.ShouldSample()` instead of invoking `_sampler.ShouldSample()` when no sampler was provided. You can do a little microbenchmark to confirm.

Reply:
I will keep this thread open and update later

Reply:
agree with @noah. Also it will be a way to check if the logger is created with sampler or not too.

Reply:
seeing AlwaysOnSampler is internal, this make my previous comment is not accurate.

Reply:
Updated